src/intervals/intervalcollection.jl

```diff
@@ -0,0 +1,245 @@
+
+
+# An IntervalCollection is an efficiently stored and indexed set of annotated
+# genomic intervals. It looks something like this.
+#
+#                                   ┌────┐            ┌────┐
+#         Each Sequence has         │chr1│            │chr2│
+#           an associated       ┌───┴────┴────┐   ┌───┴────┴────┐      ...
+#            IntervalTree       │IntervalTree │   │IntervalTree │
+#                               └─────────────┘   └─────────────┘
+#                                      │
+#                         ┌────────────┴───┬────────────┐
+#                 ╔═══════▼══════╗ ╔═══════▼══════╗     ▼
+#                 ║(10000, 20000)║ ║(35000, 40000)║
+#                 ╚══════════════╝ ╚══════════════╝     ...
+#                         │
+#    Intervals            ▼
+#     map to       ╔════════════╗
+#   liked lists    ║ Metadata 1 ║
+#   of metadata    ╚════════════╝
+#                         │
+#                         ▼
+#                  ╔════════════╗
+#                  ║ Metadata 2 ║
+#                  ╚════════════╝
+#                         │
+#                         ▼
+#
+#                        ...
+#
```

Comment:
Yeah, I was trying out [Monodraw](http://monodraw.helftone.com/). It's really nice for in-source diagrams like this. I think I may have to buy a license now that it's out of beta.